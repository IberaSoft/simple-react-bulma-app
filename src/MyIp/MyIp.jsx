import React, { Component } from 'react';

class MyIp extends Component {

    constructor(props){
        super(props);
        this.state = {
            ip: null,
            isHidden: false
        };
    }

    handleClick = (value) => {
        this.setState({ isHidden: value })
    }

    componentDidMount() {
        const url = 'https://api.ipify.org?format=json';

        fetch(url)
            .then(
                response => response.json()
            )
            .then(
                data => this.setState(data)
            );
    }

    render() {
        const { ip } = this.state;
        return (
            <div className={`notification is-warning ${(this.state.isHidden === true ) ? 'is-hidden' : ''}`}>
                <button className="delete" onClick={() => this.handleClick(true)}></button>
                { ip ? <p>Your IP is: {ip}</p> : <p>Loading...</p> }
            </div>
        )
    }

};

export default MyIp;