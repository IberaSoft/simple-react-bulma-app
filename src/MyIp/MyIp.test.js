import React from 'react';
import MyIp from './MyIp';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('MyIp component', () => {
    it('Should match the snapshot', () => {
        const renderer = new ShallowRenderer();
        const result = renderer.render(
            <MyIp />);
        expect(result).toMatchSnapshot();
    });
})