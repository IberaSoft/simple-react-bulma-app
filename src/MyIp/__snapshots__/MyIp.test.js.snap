// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyIp component Should match the snapshot 1`] = `
<div
  className="notification is-warning "
>
  <button
    className="delete"
    onClick={[Function]}
  />
  <p>
    Loading...
  </p>
</div>
`;
